File: 0801-0900/0853-Car Fleet/0853-Car Fleet.cpp
class Solution {
public:
    int carFleet(int target, vector<int>& position, vector<int>& speed) {
        map<int, double> table;
        for (int i = 0; i < position.size(); ++i) {
            table[target - position[i]] = (target - position[i]) * 1.0 / speed[i];
        }
        int count = 0;
        double t = 0;
        for (auto p : table) {            
            if (p.second > t) {
                ++count;
                t = p.second;
            }
        }
        return count;
    }
};

File: 0801-0900/0853-Car Fleet/0853-Car Fleet.py
class Solution:
    def carFleet(self, target: int, position: List[int], speed: List[int]) -> int:
        time = [(target - p) / s for p, s in sorted(zip(position, speed))]
        count = curr = 0
        for t in time[::-1]:
            if t > curr:
                count += 1
                curr = t
        return count
